pkgs/development/tools/google-java-format/default.nix

```diff
@@ -21,7 +21,12 @@ stdenv.mkDerivation rec {
     install -D ${src} $out/share/google-java-format/google-java-format.jar
 
     makeWrapper ${jre}/bin/java $out/bin/google-java-format \
-      --add-flags "-jar $out/share/google-java-format/google-java-format.jar"
+    --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
+    --add-exports jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED \
+    --add-exports jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED \
+    --add-exports jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED \
+    --add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED \
+    -jar $out/share/google-java-format/google-java-format.jar"
 
```

Comment:
This should be indented 4 spaces as is normal in shell scrips.

And it's missing `--argv0 ${pname}`.

Reply:
Actually, maybe it's better to be more clear:

```
        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED" \
        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED" \
        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED" \
        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED" \
        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED" \
        --add-flags "-jar $out/share/google-java-format/google-java-format.jar"
```

I'd rather see multiple `--add-flags` than a single string across multiple lines separated by `\`.

Reply:
(I deleted this comment by mistake, adding it again)

Actually, maybe it's better to be more clear:
```
        --argv0 ${pname} \
        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED" \
        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED" \
        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED" \
        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED" \
        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED" \
        --add-flags "-jar $out/share/google-java-format/google-java-format.jar"
```

I'd rather see multiple `--add-flags` than a single string argument over multiple lines separated by `\`.

Reply:
Ah, thanks. I wasn't aware that I could do that!

Reply:
> This should be indented 4 spaces as is normal in shell scrips.

That is wrong and only true in pure shell files. In nix files 2 spaces are always used.

Reply:
Just amended to make it 2 spaces.

Reply:
> > This should be indented 4 spaces as is normal in shell scrips.
> 
> That is wrong and only true in pure shell files. In nix files 2 spaces are always used.

Yes, Nix uses 2 spaces for indentation, including multi-line strings, lists, etc.

What's _inside_ the multi-line string is effectively a shell script, and most people switches to 4 spaces to indent the content (plus the 2 spaces of the Nix indentation), again, inside the string.

But anyway, as long it's indented it's fine, be 2, 4 or 8.